draft-ietf-quic-recovery.md

```diff
@@ -229,19 +229,23 @@ forward progress without relying on timeouts.
 ### Explicit Correction For Delayed Acknowledgements {#ack-delay}
 
 An endpoint measures the delay incurred between when a packet is received and
-when the corresponding acknowledgment is sent.  The endpoint encodes this host delay for
-the largest acknowledged packet in the Ack Delay field of an ACK frame (see
-Section 19.3 of {{QUIC-TRANSPORT}}).  This allows the receiver of the ACK to
-adjust for any host delays - importantly, for delayed acknowledgements - when
-estimating the path RTT.  Where possible, this delay can include the time that a
-packet was held in the OS kernel before being processed by a userspace QUIC
-stack.
+when the corresponding acknowledgment is sent.  The endpoint encodes this host
+delay for the largest acknowledged packet in the Ack Delay field of an ACK frame
+(see Section 19.3 of {{QUIC-TRANSPORT}}).  This allows the receiver of the ACK
+to adjust for any host delays - importantly, for delayed acknowledgements - when
+estimating the path RTT.  In certain deployments, a packet might be held in the
+OS kernel or elsewhere on the host before being processed by the QUIC
+stack. Where possible, an endpoint SHOULD include these delays when populating
+the Ack Delay field in an ACK frame.
```

Comment:
Probably not for this change, but do we really want to encourage people to be estimating delays in the kernel?  Because it seems like the point of some of the other changes is to include that sort of delay in the RTT estimate.  What matters here is end-to-end delays - the time between when the packet was consumed and when the packet containing the ACK was sent out.  Yes, it's true that you get a bigger RTT estimate if you don't correct for delays in the kernel or NIC, but those are just another piece of the plumbing now.  If we fail to cover things like scheduling delays in the RTT estimate, we're just deluding ourselves about where the delays are.

Reply:
Yeah, I was thinking about this too, but it's text that we already had. FWIW, I think it's useful to separate out these delays, not because they shouldn't get used for PTO estimates, but because knowing component delays might be helpful. That said, I don't have a good use case for these component delays yet. OTOH, I've started that separation between stack-controlled delays and other delays in the text, it might be worth using that distinction here, and arguing that these delays ought to be part of the path RTT since they're not controlled delays. I'll file a separate issue.

Reply:
Filed #2596